schedulearn-frontend/src/components/Contexts/UsernameContext.tsx

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { Component } from "react";
+
+interface UsernameProps {
+  children: React.ReactNode;
+}
+
+export interface UsernameState {
+  username: string;
+  setUsername: (username: string) => void;
+}
+
+export const UsernameContext = React.createContext<UsernameState>({
+  username: "not-from-state", 
+  setUsername: (username: string) => {console.log("Username context has not been set yet!", username);},
```

Comment:
This function should never realistically be hit as long as UsernameContext.Provider is defined in this class. In this case its probably more appropraite to throw an Error if this is ever hit.

Reply:
Added errors when functions are invoked.